### DISCUSSION TITLE
How to "decommission" a device?

### DISCUSSION THREAD

**Discussion by FezVrasta** [matter-js/matter.js#299] [Q&A] [answered]
If I pair a device with a Matter controller, and later I remove it from it (say, the remove button on Apple Home), later I can't add the device again (it will time out), what's the right process to allow a device to pair again after it's been "unpaired"?

**Comment by Apollon77** [comment, marked as answer]
Normally deleting the device in the paired controllers (apple home and such) is enough. But there is a bug in apple home especially that only one of the two added fabrics is deleted by apple right now. So after deleting device it is still considered paired. Google home do not have this issue. 

So till this is fixed the ultimate  decommissioning is to delete the storage. Use parameter for it or delete the storage dir.

**Reply by Apollon77** [reply]
Ps: for apple see also #202

**Reply by FezVrasta** [reply]
Thanks, I'm a bit confused though, Apple Home allows to delete devices even when they are unreachable. What happens in this case, especially with a "real" Matter device? Would the device stop working forever?

**Reply by Apollon77** [reply]
You simply do a factory reset of the device in this case ... in fact deleting the storage in matter.js case is "the same"

**Reply by FezVrasta** [reply]
Thanks. If I have multiple devices all using the same storage how can I delete from it just the data about that specific device? I tried to delete the files with the device ID in them but it doesn't seem to be enough.

**Reply by Apollon77** [reply]
What exactly you mean with multiple devices? A bridge or what?

**Reply by FezVrasta** [reply]
> What exactly you mean with multiple devices? A bridge or what?

Multiple OnOff devices all attached to the same matter server (each one with different port number and commissioning server)

**Reply by Apollon77** [reply]
When you check the matter.js roadmap you will find an entry to allow clean removals if such devices and such. Right now i
Can not tell for sure what's all needed. 

Please be aware that matter.js is still under heavy development in some high level api regards. This is one of them. 
I'm theory or should work with calling close/stop on the CommissioningServer instance but I fear because it uses the "array index" as key right now (which might no have been the best idea) that on next start when you add the devices without this one it screwes up all devices. 

So that being said there might come a breaking change as soon as unclean that up. Maybe I find a better solution. But not sure yet
